Feature/viewsets

zhenyabigel opened the pull request:
implement model APIs with tests, Add Model ViewSets with proper mixins, Configure permissions and routers - Add filtering (django-filters), Write comprehensive tests

Garyyys commented on autodealer_backend/cars/filters.py:
```diff
@@ -0,0 +1,69 @@
+class CarFilter(django_filters.FilterSet):
+    brand = django_filters.CharFilter(lookup_expr='icontains')
+    model = django_filters.CharFilter(lookup_expr='icontains')
+    year__gt = django_filters.NumberFilter(field_name='year', lookup_expr='gt')
+    year__lt = django_filters.NumberFilter(field_name='year', lookup_expr='lt')
+    price__gt = django_filters.NumberFilter(field_name='price', lookup_expr='gt')
+    price__lt = django_filters.NumberFilter(field_name='price', lookup_expr='lt')
```
Мб добавить gte/lte

Garyyys commented on autodealer_backend/cars/models.py:
```diff
@@ -0,0 +1,72 @@
+from django.db import models
+
+
+class Supplier(models.Model):
```
А почему в этот MR доехали модели ?

Garyyys commented on autodealer_backend/dealers/serializers.py:
```diff
@@ -0,0 +1,32 @@
+class DealerSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Dealer
+        fields = '__all__'
```
В принципе плохо отдавать абсолютно все, в реальном проекте будет сказываться на перформансе, но это на будущее

Garyyys commented on autodealer_backend/dealers/views.py:
```diff
@@ -0,0 +1,70 @@
+class DealerViewSet(mixins.ListModelMixin,
+                    mixins.RetrieveModelMixin,
+                    mixins.UpdateModelMixin,
+                    GenericViewSet):
```
странное форматирование, ждем pre-commit

Garyyys commented on autodealer_backend/users/authentication.py:
```diff
@@ -0,0 +1,25 @@
+class LogoutView(APIView):
+    permission_classes = [AllowAny]
+
+    def post(self, request):
+        try:
+            refresh_token = request.data["refresh_token"]
+            token = RefreshToken(refresh_token)
+            token.blacklist()
+            return Response(status=status.HTTP_205_RESET_CONTENT)
+        except Exception as e:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
```
Нам точно тут нужен APIView?

Garyyys commented:
переназначь target branch на ветку с которой делал эту

zhenyabigel committed:
feat:
Add ViewSets with proper mixins,
Configure permissions and routers
Add filtering (django-filters)

zhenyabigel committed:
fix: review mistakes
